<html><h1>Validation FAILED.</h1><p>Build Time: 14 minutes 15 seconds 138 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/053db72762d6527d7019878d7b655ec5dbd256d0">053db72762d6527d7019878d7b655ec5dbd256d0</a></p><h6>Task Summary:</h6><ul><li>subrepository-compile-jdk8 -  :white_check_mark:</li><li>subrepository-semantic-versioning-jdk8 -  :x:<pre><code>     [echo] :apps:commerce:commerce-product-tax-category-web:mergeLang
     [echo] :apps:commerce:commerce-product-tax-category-web:replaceSoyTranslation UP-TO-DATE
     [echo] :apps:commerce:commerce-product-tax-category-web:wrapSoyAlloyTemplate SKIPPED
     [echo] :apps:commerce:commerce-product-tax-category-web:classes
     [echo] :apps:commerce:commerce-product-tax-category-web:jar
     [echo] :apps:commerce:commerce-product-tax-category-web:baseline
     [echo] :apps:commerce:commerce-product-tax-category-web:syncVersions
     [echo] :apps:commerce:commerce-product-test:baseline SKIPPED
     [echo] :apps:commerce:commerce-product-test:syncVersions
     [echo] :apps:commerce:commerce-product-test-util:compileJava UP-TO-DATE
     [echo] :apps:commerce:commerce-product-test-util:buildCSS UP-TO-DATE
     [echo] :apps:commerce:commerce-product-test-util:processResources
     [echo] :apps:commerce:commerce-product-test-util:transpileJS SKIPPED
     [echo] :apps:commerce:commerce-product-test-util:configJSModules SKIPPED
     [echo] :apps:commerce:commerce-product-test-util:copyLibs SKIPPED
     [echo] :apps:commerce:commerce-product-test-util:replaceSoyTranslation UP-TO-DATE
     [echo] :apps:commerce:commerce-product-test-util:wrapSoyAlloyTemplate SKIPPED
     [echo] :apps:commerce:commerce-product-test-util:classes
     [echo] :apps:commerce:commerce-product-test-util:jar
     [echo] :apps:commerce:commerce-product-test-util:baseline
     [echo] [Baseline Report] Mode: diff (persisted)
     [echo] [Baseline Warning] Bundle Version Change Recommended: 1.1.0
     [echo]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [echo] = ================================================== ========== ========== ========== ========== ==========
     [echo] * com.liferay.commerce.product.test.util             MINOR      1.0.0      1.0.0      1.1.0      VERSION INCREASE REQUIRED
     [echo] 	&lt;   class      com.liferay.commerce.product.test.util.CPTestUtil
     [echo] 		+   method     getSearchContext(java.lang.String,int,com.liferay.portal.kernel.model.Group)
     [echo] 			+   access     static
     [echo] 			+   return     com.liferay.portal.kernel.search.SearchContext
     [echo] :apps:commerce:commerce-product-test-util:baseline FAILED
     [echo] :apps:commerce:commerce-product-test-util:syncVersions
     [echo] 
     [echo] BUILD FAILED
     [echo] 
     [echo] </code></pre></li></ul><h5>For full console, click <a href="https://test-1-19.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/6//consoleText">here</a>.</h5><hr/><div><h6>Test Results:</h6><h5>No tests were run.</h5></div></html>

--- a/commerce-product-test-util/bnd.bnd
+++ b/commerce-product-test-util/bnd.bnd
@@ -1,4 +1,4 @@
 Bundle-Name: Liferay Commerce Product Test Utilities
 Bundle-SymbolicName: com.liferay.commerce.product.test.util
-Bundle-Version: 1.0.3
+Bundle-Version: 1.1.0
 Export-Package: com.liferay.commerce.product.test.util

--- a/commerce-product-test-util/build.gradle
+++ b/commerce-product-test-util/build.gradle
@@ -1,4 +1,5 @@
 dependencies {
+	compileOnly group: "com.liferay", name: "com.liferay.petra.string", version: "2.0.0"
 	compileOnly group: "com.liferay.commerce", name: "com.liferay.commerce.product.type.simple", version: "1.0.0"
 	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "3.5.0"
 	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.test", version: "1.0.0"

--- a/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java
+++ b/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java
@@ -32,9 +32,13 @@ import com.liferay.commerce.product.service.CPInstanceLocalServiceUtil;
 import com.liferay.commerce.product.service.CPOptionLocalServiceUtil;
 import com.liferay.commerce.product.service.CPOptionValueLocalServiceUtil;
 import com.liferay.commerce.product.type.simple.constants.SimpleCPTypeConstants;
+import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Group;
 import com.liferay.portal.kernel.model.User;
 import com.liferay.portal.kernel.module.configuration.ConfigurationProviderUtil;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.SearchContext;
 import com.liferay.portal.kernel.service.ServiceContext;
 import com.liferay.portal.kernel.service.UserLocalServiceUtil;
 import com.liferay.portal.kernel.settings.SystemSettingsLocator;
@@ -42,14 +46,20 @@ import com.liferay.portal.kernel.test.util.RandomTestUtil;
 import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
 import com.liferay.portal.kernel.util.CalendarFactoryUtil;
 import com.liferay.portal.kernel.util.Time;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.io.Serializable;
 
 import java.util.Calendar;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Locale;
 import java.util.Map;
 
 /**
  * @author Andrea Di Giorgi
+ * @author Luca Pellizzon
  */
 public class CPTestUtil {
 
@@ -138,8 +148,7 @@ public class CPTestUtil {
 			ServiceContextTestUtil.getServiceContext(groupId);
 
 		CPDefinition cpDefinition = _addCPDefinition(
-			SimpleCPTypeConstants.NAME, RandomTestUtil.randomBoolean(), true,
-			serviceContext);
+			SimpleCPTypeConstants.NAME, true, true, serviceContext);
 
 		return CPInstanceLocalServiceUtil.getCPInstance(
 			cpDefinition.getCPDefinitionId(), CPInstanceConstants.DEFAULT_SKU);
@@ -188,6 +197,41 @@ public class CPTestUtil {
 			cpDefinition.getCPDefinitionId(), serviceContext);
 	}
 
+	public static SearchContext getSearchContext(
+		String keywords, int status, Group group) {
+
+		SearchContext searchContext = new SearchContext();
+
+		Map<String, Serializable> attributes = new HashMap<>();
+
+		attributes.put(Field.STATUS, status);
+
+		LinkedHashMap<String, Object> params = new LinkedHashMap<>();
+
+		if (Validator.isNotNull(keywords)) {
+			params.put("keywords", keywords);
+		}
+		else {
+			params.put("keywords", StringPool.STAR);
+		}
+
+		attributes.put("params", params);
+
+		searchContext.setAttributes(attributes);
+
+		searchContext.setCompanyId(group.getCompanyId());
+		searchContext.setGroupIds(new long[] {group.getGroupId()});
+
+		if (Validator.isNotNull(keywords)) {
+			searchContext.setKeywords(keywords);
+		}
+		else {
+			searchContext.setKeywords(StringPool.STAR);
+		}
+
+		return searchContext;
+	}
+
 	private static CPDefinition _addCPDefinition(
 			String productTypeName, boolean ignoreSKUCombinations,
 			boolean hasDefaultInstance, ServiceContext serviceContext)
@@ -222,7 +266,7 @@ public class CPTestUtil {
 		boolean taxExempt = RandomTestUtil.randomBoolean();
 		boolean telcoOrElectronics = RandomTestUtil.randomBoolean();
 		String ddmStructureKey = null;
-		boolean published = RandomTestUtil.randomBoolean();
+		boolean published = true;
 
 		Date displayDate = new Date(now - Time.HOUR);
 		Date expirationDate = new Date(now + Time.DAY);

--- a/commerce-product-test-util/src/main/resources/com/liferay/commerce/product/test/util/packageinfo
+++ b/commerce-product-test-util/src/main/resources/com/liferay/commerce/product/test/util/packageinfo
@@ -1,1 +1,1 @@
-version 1.0.0
+version 1.1.0

--- a/commerce-product-test/build.gradle
+++ b/commerce-product-test/build.gradle
@@ -6,7 +6,6 @@ dependencies {
 	compileOnly group: "com.github.ignaciotcrespo", name: "frutilla", version: "0.7.1"
 
 	testIntegrationCompile group: "com.liferay", name: "com.liferay.arquillian.extension.junit.bridge", version: "1.0.7"
-	testIntegrationCompile group: "com.liferay", name: "com.liferay.petra.string", version: "2.0.0"
 	testIntegrationCompile group: "com.liferay", name: "com.liferay.portal.search.test.util", version: "2.1.0"
 	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "3.5.0"
 	testIntegrationCompile project(":apps:commerce:commerce-product-api")

--- a/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/util/test/CPDefinitionHelperTest.java
+++ b/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/util/test/CPDefinitionHelperTest.java
@@ -23,37 +23,28 @@ import com.liferay.commerce.product.model.CPDefinition;
 import com.liferay.commerce.product.model.CPInstance;
 import com.liferay.commerce.product.test.util.CPTestUtil;
 import com.liferay.commerce.product.util.CPDefinitionHelper;
-import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.dao.orm.QueryUtil;
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.model.Group;
-import com.liferay.portal.kernel.search.Field;
 import com.liferay.portal.kernel.search.SearchContext;
 import com.liferay.portal.kernel.test.rule.AggregateTestRule;
 import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
 import com.liferay.portal.kernel.test.util.GroupTestUtil;
 import com.liferay.portal.kernel.util.ArrayUtil;
 import com.liferay.portal.kernel.util.ListUtil;
-import com.liferay.portal.kernel.util.Validator;
 import com.liferay.portal.kernel.workflow.WorkflowConstants;
 import com.liferay.portal.test.rule.Inject;
 import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
 import com.liferay.portal.test.rule.PermissionCheckerTestRule;
 
-import java.io.Serializable;
-
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Map;
 
 import org.frutilla.FrutillaRule;
 
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.ClassRule;
-import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -76,7 +67,6 @@ public class CPDefinitionHelperTest {
 		_group = GroupTestUtil.addGroup();
 	}
 
-	@Ignore
 	@Test
 	public void testSearchCategory() throws PortalException {
 		frutillaRule.scenario(
@@ -116,8 +106,8 @@ public class CPDefinitionHelperTest {
 		AssetCategory assetCategory = CPTestUtil.addCategoryToCPDefinitions(
 			_group.getGroupId(), cpDefinitionIds);
 
-		SearchContext searchContext = _getSearchContext(
-			null, WorkflowConstants.STATUS_APPROVED);
+		SearchContext searchContext = CPTestUtil.getSearchContext(
+			null, WorkflowConstants.STATUS_APPROVED, _group);
 
 		CPQuery cpQuery = new CPQuery();
 
@@ -150,38 +140,6 @@ public class CPDefinitionHelperTest {
 	}
 
 	@Test
-	public void testSearchDraft() throws PortalException {
-		frutillaRule.scenario(
-			"Search for CPDefinitions by 'draft' status"
-		).given(
-			"A collection of CPDefinitions"
-		).when(
-			"I search for CPDefinitions given the status as a search filter"
-		).then(
-			"The results will contain only those product in draft status. In " +
-				"this specific case 0"
-		);
-
-		_addCPInstances(_group.getGroupId(), _CP_INSTANCES_COUNT);
-
-		SearchContext searchContext = _getSearchContext(
-			null, WorkflowConstants.STATUS_DRAFT);
-
-		CPQuery cpQuery = new CPQuery();
-
-		CPDataSourceResult cpDataSourceResult = _cpDefinitionHelper.search(
-			_group.getGroupId(), searchContext, cpQuery, QueryUtil.ALL_POS,
-			QueryUtil.ALL_POS);
-
-		List<CPCatalogEntry> cpCatalogEntries =
-			cpDataSourceResult.getCPCatalogEntries();
-
-		Assert.assertEquals(
-			cpCatalogEntries.toString(), 0, cpCatalogEntries.size());
-	}
-
-	@Ignore
-	@Test
 	public void testSearchName() throws PortalException {
 		frutillaRule.scenario(
 			"Search for CPDefinitions by name"
@@ -203,8 +161,8 @@ public class CPDefinitionHelperTest {
 
 		CPDefinition cpDefinition = randomCPInstance.getCPDefinition();
 
-		SearchContext searchContext = _getSearchContext(
-			cpDefinition.getName(), WorkflowConstants.STATUS_APPROVED);
+		SearchContext searchContext = CPTestUtil.getSearchContext(
+			cpDefinition.getName(), WorkflowConstants.STATUS_APPROVED, _group);
 
 		CPQuery cpQuery = new CPQuery();
 
@@ -225,7 +183,6 @@ public class CPDefinitionHelperTest {
 			cpCatalogEntry.getCPDefinitionId());
 	}
 
-	@Ignore
 	@Test
 	public void testSearchStar() throws PortalException {
 		frutillaRule.scenario(
@@ -241,8 +198,8 @@ public class CPDefinitionHelperTest {
 		CPInstance[] cpInstances = _addCPInstances(
 			_group.getGroupId(), _CP_INSTANCES_COUNT);
 
-		SearchContext searchContext = _getSearchContext(
-			null, WorkflowConstants.STATUS_APPROVED);
+		SearchContext searchContext = CPTestUtil.getSearchContext(
+			null, WorkflowConstants.STATUS_APPROVED, _group);
 
 		CPQuery cpQuery = new CPQuery();
 
@@ -283,38 +240,6 @@ public class CPDefinitionHelperTest {
 		}
 
 		return cpInstances;
-	}
-
-	private SearchContext _getSearchContext(String keywords, int status) {
-		SearchContext searchContext = new SearchContext();
-
-		LinkedHashMap<String, Object> params = new LinkedHashMap<>();
-
-		if (Validator.isNotNull(keywords)) {
-			params.put("keywords", keywords);
-		}
-		else {
-			params.put("keywords", StringPool.STAR);
-		}
-
-		Map<String, Serializable> attributes = new HashMap<>();
-
-		attributes.put(Field.STATUS, status);
-		attributes.put("params", params);
-
-		searchContext.setAttributes(attributes);
-
-		searchContext.setCompanyId(_group.getCompanyId());
-		searchContext.setGroupIds(new long[] {_group.getGroupId()});
-
-		if (Validator.isNotNull(keywords)) {
-			searchContext.setKeywords(keywords);
-		}
-		else {
-			searchContext.setKeywords(StringPool.STAR);
-		}
-
-		return searchContext;
 	}
 
 	private static final int _CP_INSTANCES_COUNT = 10;

--- a/commerce-test/src/testIntegration/java/com/liferay/commerce/product/rule/test/util/CPRuleTestUtil.java
+++ b/commerce-test/src/testIntegration/java/com/liferay/commerce/product/rule/test/util/CPRuleTestUtil.java
@@ -104,14 +104,6 @@ public class CPRuleTestUtil {
 			commerceUserSegmentEntry.getCommerceUserSegmentEntryId(),
 			serviceContext);
 
-		CommerceUserSegmentEntry defaultCommerceUserSegmentEntry =
-			CommerceUserSegmentTestUtil.addDefaultUserSegment(groupId);
-
-		CPRuleUserSegmentRelLocalServiceUtil.addCPRuleUserSegmentRel(
-			cpRule.getCPRuleId(),
-			defaultCommerceUserSegmentEntry.getCommerceUserSegmentEntryId(),
-			serviceContext);
-
 		return cpRule;
 	}
 

--- a/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CPRuleTest.java
+++ b/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CPRuleTest.java
@@ -1,0 +1,222 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.asset.kernel.model.AssetCategory;
+import com.liferay.commerce.product.catalog.CPCatalogEntry;
+import com.liferay.commerce.product.catalog.CPQuery;
+import com.liferay.commerce.product.data.source.CPDataSourceResult;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.model.CPInstance;
+import com.liferay.commerce.product.model.CPRule;
+import com.liferay.commerce.product.rule.test.util.CPRuleTestUtil;
+import com.liferay.commerce.product.service.CPDefinitionService;
+import com.liferay.commerce.product.test.util.CPTestUtil;
+import com.liferay.commerce.product.util.CPDefinitionHelper;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionCheckerFactoryUtil;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
+
+import java.util.List;
+
+import org.frutilla.FrutillaRule;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Luca Pellizzon
+ */
+@RunWith(Arquillian.class)
+public class CPRuleTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_user1 = UserTestUtil.addUser();
+
+		_user2 = UserTestUtil.addUser();
+
+		PermissionChecker permissionChecker =
+			PermissionCheckerFactoryUtil.create(_user1);
+
+		PermissionThreadLocal.setPermissionChecker(permissionChecker);
+	}
+
+	@Test
+	public void testUserSegmentCategoryCPDefinitionHelper() throws Exception {
+		frutillaRule.scenario(
+			"Search for CatalogEntries accessible from a given user"
+		).given(
+			"A user, a category with a product"
+		).when(
+			"I try to get the CatalogEntries that the user can view"
+		).then(
+			"The results will contain only accessible element from that " +
+				"specific user"
+		);
+
+		CPInstance cpInstance = CPTestUtil.addCPInstance(_group.getGroupId());
+
+		CPTestUtil.addCPInstance(_group.getGroupId());
+		CPTestUtil.addCPInstance(_group.getGroupId());
+
+		CPDefinition expectedCPDefinition = cpInstance.getCPDefinition();
+
+		CPRule cpRule = CPRuleTestUtil.addUserSegmentCPRule(
+			_group.getGroupId(), _user1.getUserId());
+
+		AssetCategory assetCategory = CPRuleTestUtil.addAssetCategoryToCPRule(
+			cpRule);
+
+		CPRuleTestUtil.setCPRulesThreadLocal(
+			_group.getGroupId(), 0, _user1.getUserId());
+
+		CPRuleTestUtil.addCategoryToCPDefinition(
+			expectedCPDefinition, assetCategory.getCategoryId());
+
+		SearchContext searchContext = CPTestUtil.getSearchContext(
+			null, WorkflowConstants.STATUS_APPROVED, _group);
+
+		CPQuery cpQuery = new CPQuery();
+
+		cpQuery.setAllCategoryIds(new long[] {assetCategory.getCategoryId()});
+
+		CPDataSourceResult cpDataSourceResult = _cpDefinitionHelper.search(
+			_group.getGroupId(), searchContext, cpQuery, QueryUtil.ALL_POS,
+			QueryUtil.ALL_POS);
+
+		List<CPCatalogEntry> cpCatalogEntries =
+			cpDataSourceResult.getCPCatalogEntries();
+
+		Assert.assertEquals(
+			cpCatalogEntries.toString(), 1, cpCatalogEntries.size());
+	}
+
+	@Test
+	public void testUserSegmentCategoryCPRuleService() throws Exception {
+		frutillaRule.scenario(
+			"Get a CatalogEntries accessible from a given user"
+		).given(
+			"A user, a category with a product"
+		).when(
+			"I try to get the CatalogEntries that the user can view"
+		).then(
+			"The results will contain the accessible element from that " +
+				"specific user"
+		);
+
+		CPInstance cpInstance = CPTestUtil.addCPInstance(_group.getGroupId());
+
+		CPDefinition expectedCPDefinition = cpInstance.getCPDefinition();
+
+		CPRule cpRule = CPRuleTestUtil.addUserSegmentCPRule(
+			_group.getGroupId(), _user1.getUserId());
+
+		AssetCategory assetCategory = CPRuleTestUtil.addAssetCategoryToCPRule(
+			cpRule);
+
+		CPRuleTestUtil.setCPRulesThreadLocal(
+			_group.getGroupId(), 0, _user1.getUserId());
+
+		CPRuleTestUtil.addCategoryToCPDefinition(
+			expectedCPDefinition, assetCategory.getCategoryId());
+
+		CPDefinition actualCPDefinition = _cpDefinitionService.getCPDefinition(
+			expectedCPDefinition.getCPDefinitionId());
+
+		Assert.assertEquals(
+			expectedCPDefinition.getCPDefinitionId(),
+			actualCPDefinition.getCPDefinitionId());
+	}
+
+	@Test(expected = PrincipalException.class)
+	public void testWrongUserSegmentCategoryCPRuleService() throws Exception {
+		frutillaRule.scenario(
+			"Try to get a CatalogEntries not accessible from a given user"
+		).given(
+			"A user, a category with a product"
+		).when(
+			"I try to get the CatalogEntries that the user cannot view"
+		).then(
+			"The results will be a PrincipalException caused by the " +
+				"permission checks"
+		);
+
+		CPInstance cpInstance = CPTestUtil.addCPInstance(_group.getGroupId());
+
+		CPDefinition expectedCPDefinition = cpInstance.getCPDefinition();
+
+		CPRule cpRule = CPRuleTestUtil.addUserSegmentCPRule(
+			_group.getGroupId(), _user2.getUserId());
+
+		AssetCategory assetCategory = CPRuleTestUtil.addAssetCategoryToCPRule(
+			cpRule);
+
+		CPRuleTestUtil.setCPRulesThreadLocal(
+			_group.getGroupId(), 0, _user1.getUserId());
+
+		CPRuleTestUtil.addCategoryToCPDefinition(
+			expectedCPDefinition, assetCategory.getCategoryId());
+
+		_cpDefinitionService.getCPDefinition(
+			expectedCPDefinition.getCPDefinitionId());
+	}
+
+	@Rule
+	public FrutillaRule frutillaRule = new FrutillaRule();
+
+	@Inject
+	private CPDefinitionHelper _cpDefinitionHelper;
+
+	@Inject
+	private CPDefinitionService _cpDefinitionService;
+
+	@DeleteAfterTestRun
+	private Group _group;
+
+	@DeleteAfterTestRun
+	private User _user1;
+
+	@DeleteAfterTestRun
+	private User _user2;
+
+}

